fatiando/mesher.py

```diff
@@ -1721,3 +1721,594 @@ def keep(cell):
+    incorporating remanence, demagnetization and anisotropy: HP 41C
+    handheld computer algorithms, Exploration Geophysics, 16, 1-122, 1985.
+
+    .. note:: The coordinate system used is x -> North, y -> East
+    and z -> Down
+
+    Parameters:
+
+    * x, y, z : float
+        The coordinates of the center of the ellipsoid.
+    * large_axis, small_axis: float
+        Semi-axes forming the ellipsoid.
+    * strike, dip, rake
+        Orientation angles of the ellipsoid.
+    * props : dict
+        Physical properties assigned to the ellipsoid.
+        Ex: ``props={'density':10,
+                     'remanent magnetization':[13, -5, 7.4],
+                     'susceptibility tensor':[0.562, 0.485, 0.25,
+                                              0, 7, 29.4]}``
+
+    Examples:
+
+        >>> e = ProlateEllipsoid(x=1, y=2, z=3, large_axis=6, small_axis=4,
+        ...                      strike=10, dip=20, rake=30, props={
+        ...                      'remanent magnetization': [10, 25, 40],
+        ...                      'susceptibility tensor': [0.562, 0.485,
+        ...                                                0.25, 90, 0,
+        ...                                                0]})
+        >>> e.props['remanent magnetization']
+        [10, 25, 40]
+        >>> e.addprop('density', 20)
+        >>> print(e.props['density'])
+        20
+        >>> print(e)
+        x:1 | y:2 | z:3 | large_axis:6 | small_axis:4 | strike:10 | dip:20 | r\
+ake:30 | density:20 | remanent magnetization:[10, 25, 40] | susceptibility ten\
+sor:[0.562, 0.485, 0.25, 90, 0, 0]
+        >>> e = ProlateEllipsoid(1, 2, 3, 6, 4, 10, 20, 30)
+        >>> print(e)
+        x:1 | y:2 | z:3 | large_axis:6 | small_axis:4 | strike:10 | dip:20 | r\
+ake:30
+        >>> e.addprop('density', 2670)
+        >>> print(e)
+        x:1 | y:2 | z:3 | large_axis:6 | small_axis:4 | strike:10 | dip:20 | r\
+ake:30 | density:2670
+
+    """
+
+    def __init__(self, x, y, z, large_axis, small_axis,
+                 strike, dip, rake, props=None):
+        GeometricElement.__init__(self, props)
+
+        self.x = x
+        self.y = y
+        self.z = z
+        self.large_axis = large_axis
+        self.small_axis = small_axis
+        self.strike = strike
+        self.dip = dip
+        self.rake = rake
+
+        assert self.large_axis > self.small_axis, "large_axis must be greater \
+than small_axis"
+
+        # Auxiliary orientation angles
+        self.alpha, self.gamma, self.delta = auxiliary_angles(self.strike,
+                                                              self.dip,
+                                                              self.rake)
+
+        # Coordinate transformation matrix
+        self.transf_matrix = coord_transf_matrix_triaxial(self.alpha,
+                                                          self.gamma,
+                                                          self.delta)
```

Comment:
Actually, only the `transf_matrix` is used in the forward modelling. I will remove the `self.` from `alpha`, `gamma` and `delta` .